components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java

```diff
@@ -115,6 +122,56 @@ public ApplicationModel getApplication(String applicationId) {
         }
     }
 
+    /**
+     * Export an application identified by the applicationId, as an XML string.
+     *
+     * @param applicationId ID of the application to be exported.
+     * @param exportSecrets If True, all hashed or encrypted secrets will also be exported.
+     * @return XML string of the application.
+     */
+    public String exportApplication(String applicationId, Boolean exportSecrets) {
+
+        try {
+            // Pull application using the ID to get the App name
+            String tenantDomain = ContextLoader.getTenantDomainFromContext();
+            ApplicationBasicInfo application =
+                    getApplicationManagementService().getApplicationBasicInfoByResourceId(applicationId, tenantDomain);
+            if (application == null) {
+                throw buildApiError(ErrorMessage.ERROR_CODE_APPLICATION_NOT_FOUND, applicationId, tenantDomain);
+            }
+            String appName = application.getApplicationName();
+            // Pass app name and get the XML output
+            return getApplicationManagementService().exportSPApplication(appName, exportSecrets, tenantDomain);
+        } catch (IdentityApplicationManagementException e) {
+            throw handleServerError(e, "Error while retrieving application with id: " + applicationId);
+        }
+    }
+
+    public ApplicationModel importApplication(InputStream fileInputStream, Attachment fileDetail) {
+
+        try {
+            SpFileContent spFileContent = new SpFileContent();
+            spFileContent.setContent(IOUtils.toString(fileInputStream, StandardCharsets.UTF_8.name()));
+            spFileContent.setFileName(fileDetail.getDataHandler().getName());
+
+            String tenantDomain = ContextLoader.getTenantDomainFromContext();
+            String username = ContextLoader.getUsernameFromContext();
+            ImportResponse importResponse = getApplicationManagementService().importSPApplication(
+                    spFileContent, tenantDomain, username, false);
+            if (importResponse.getResponseCode() == ImportResponse.CREATED) {
+                ServiceProvider application =
+                        getApplicationManagementService().getApplicationExcludingFileBasedSPs(
+                                importResponse.getApplicationName(), tenantDomain);
+                return new ServiceProviderToApiModel().apply(application);
+            } else {
+                throw buildApiError(ErrorMessage.ERROR_IMPORTING_APPLICATION);
+            }
+        } catch (IOException | IdentityApplicationManagementException e) {
+            // TODO: 2019-11-08 need to handle client error once Framework changes are merged.
+            throw handleServerError(e, "Error while importing application from XML file.");
```

Comment:
Shall we implement that logic here? So that automatically client errors are detected and thrown with 400.